### DISCUSSION TITLE
Can't get reactivity to work on page refresh with useFetch

### DISCUSSION THREAD

**Discussion by tchartron** [nuxt/nuxt#30196] [Questions] [answered]
Hi everyone, i need help i must be doing something wrong but i can't figure out what.
I tried to create a reproduction on stackblitz but i wasn't able to ... 

Here is the (simplified) scenario : **I have an edit page, at the root of `script setup` i use `useFetch` to get some data from an api, then i assign the value to a `form` ref() in order to use it with the form inputs v-models.**

**The problem :** 
When navigating (using `<NuxtLink :to="/website/edit/1" />`) to the page data is fetched, inputs are populated normally.
But when refreshing the page or accessing it directly from url in the browser, data is fetched, the `website` ref() is populated but reactivity is not triggered for the `form` ref() and it stays undefined, resulting in form inputs not being populated.

I tried using `useAsyncData` and i had the same result.

However when using `$fetch` instead, it works just fine.

Here is a simplified example **not working** on page refresh `/pages/website/edit/[id].vue` : 
```vue
<script setup lang="ts">
type Metadata = {
  title: string;
  description: string;
}
interface Website {
  name: string;
  metadata: Metadata;
}

const { data: website, status } = await useFetch<Website>(`/websites/${route.params.id}`, {
    key: `website-${route.params.id}`,
    lazy: true,
    baseURL: runtimeConfig.public.apiUrl,
    credentials: 'include',
    headers: {
        'Accept': 'application/json',
        'Referer': 'https://dash.test.docker'
        'x-locale': useNuxtApp().$i18n.locale.value,
    },
    watch: false
})

const form = ref<EditWebsiteForm>({
    name: website.value?.name as string,
    metadata: website.value?.metadata as object,
})
</script>

<template>
    <input v-model="form.name" type="text" placeholder="website name" />
    <input v-model="form.metadata.title" type="text" placeholder="website title" />
</template>
```
____

Here is the same example **that works** on page refresh `/pages/website/edit/[id].vue` : 
```vue
<script setup lang="ts">
type Metadata = {
  title: string;
  description: string;
}
interface Website {
  name: string;
  metadata: Metadata;
}

await $fetch(`/websites/${route.params.id}`, {
    baseURL: runtimeConfig.public.apiUrl,
    credentials: 'include',
    headers: {
        'Accept': 'application/json',
        'Referer': 'https://dash.test.docker'
        'x-locale': useNuxtApp().$i18n.locale.value,
    }
}).then((response) => {
     website.value = response as Website
})

const form = ref<EditWebsiteForm>({
    name: website.value?.name as string,
    metadata: website.value?.metadata as object,
})
</script>

<template>
    <input :value="form.name" type="text" placeholder="website name" />
    <input :value="form.metadata.title" type="text" placeholder="website title" />
</template>
```

I tried to simplify the code as much as i could to illustrate the problem i'm facing, thanks a lot for taking the time to point me in a direction.

**Comment by xjccc** [comment]
If you set `lazy: true`, it will fetch data on mounted.

I think you should use `watch`

```
watch(() => website.value, () => {
   form.value.name = website.value.name
   form.value.metadata = website.value.metadata
})
```

**Comment by tchartron** [comment]
Hey @xjccc thanks for helping, watch works indeed but not for nested objects, for example it does not work for 
```html
<input :value="form.metadata.title" type="text" placeholder="website title" />
```

Even with the `{ deep: true }` option specified on the watcher.
And it feels hacky doesn't it ?

**Reply by xjccc** [reply]
I think remove  `lazy` will fix it?

**Reply by tchartron** [reply]
No, sorry i tried it of course, also tried with `server: false` and other useFetch options ...

**Reply by xjccc** [reply]
emm.. I test this, whether jump from front page or refresh, `form` alse has value?

https://stackblitz.com/edit/nuxt-starter-ywvw9eoq?file=app.vue,pages%2Findex.vue,pages%2F[id].vue,server%2Fapi%2Fid.ts

**Reply by tchartron** [reply]
This one does not work on my setup either, but i must be doing something wrong elsewhere since we are not able to reproduce on stackblitz

**Reply by tchartron** [reply]
I ended up using the watcher as suggested, feels a bit hacky but i guess it's not, thanks for helping @xjccc (the nested object problem mentionned earlier was due to something else, it's indeed working as expected now)

**Reply by tchartron** [reply, marked as answer]
Well i've found what was causing this behaviour ... I was wrapping the whole app (at layout level) into the `<ColorScheme>` component [from here](https://github.com/nuxt-modules/color-mode/blob/main/src/runtime/component.vue3.vue), which wraps the whole app into the nuxt `<ClientOnly>` component.
I'm adding this in case it helps someone later, thanks for helping @xjccc  👍🏼
